crates/subspace-node/src/commands/run/consensus.rs

```diff
@@ -387,16 +394,60 @@ pub(super) struct ConsensusChainOptions {
     #[clap(flatten)]
     dsn_options: DsnOptions,
 
-    /// Enables DSN-sync on startup.
-    #[arg(long, default_value_t = true, action = clap::ArgAction::Set)]
-    sync_from_dsn: bool,
-
     /// Parameters used to create the storage monitor.
     #[clap(flatten)]
     storage_monitor: StorageMonitorParams,
 
     #[clap(flatten)]
     timekeeper_options: TimekeeperOptions,
+
+    /// Sync mode
+    #[arg(long, default_value_t = ConfigSyncMode::Full)]
+    sync: ConfigSyncMode,
+}
+
+/// Syncing mode.
+#[derive(Debug, Clone, Copy, PartialEq, Parser)]
+pub enum ConfigSyncMode {
+    /// Full sync. Download and verify all blocks.
+    Full,
+    /// Download blocks from DSN.
+    Dsn,
```

Comment:
I'm not sure why we have these as separate modes, nothing in the code seems to make any distinction between them. Also we don't need separate `ConfigSyncMode` and `ChainSyncMode`, just `ChainSyncMode` that implements `FromStr` is sufficient.

Reply:
We use `Snap` and `not Full`, so we use `Dsn` implicitly. Two types `ChainSyncMode` and `ConfigSyncMode` were introduced to avoid adding `clap` dependency into `subspace-service`.

Reply:
I didn't understand what you wanted to say with the first sentence, can you rephrase?

As for Clap, you don't need it. Enum only needs to implement `FromStr` trait IIRC.

Reply:
We use `Dsn` value implicitly without a code reference as following: we either use only `Full` (regular) sync or DSN sync option when we can run `Snap` sync first.

> As for Clap, you don't need it. Enum only needs to implement FromStr trait IIRC.

Indeed!

Reply:
> We use Dsn value implicitly without a code reference as following: we either use only Full (regular) sync or DSN sync option when we can run Snap sync first.

I'm sorry, it still makes no sense to me, at least it doesn't seem to answer the question I had originally.

Reply:
Previously, we had two sync modes - regular and DSN, the latter was enabled by `--sync-from-dsn` and started a dedicated task.
Now, we have three modes: full, dsn, and snap. Snap starts in the same dedicated task as Dsn sync. 
We start the dedicated task by checking `if !config.sync.is_full()`. "Not Full" includes Dsn and Snap sync modes. Snap sync can be activated by `Snap` sync mode before the Dsn worker in the dedicated sync task. 
Thus we use Dsn sync mode implicitly without referencing it in the code. 

I hope it makes more sense now. Feel free to ask additional questions.

Reply:
I missed `is_full` check somehow. You made `Full` sync the default, which will disable DSN sync if I understand correctly. I'd say we should simply remove ability to disable sync from DSN. There should be just `Full` and `Snap`. No need to have `Dsn` as a third option, users will not be able to sync fully without DSN anyway (almost all nodes are pruned).

Reply:
Will domain operators be affected by this change? We're changing our previous semantics.

Reply:
What semantics? DSN sync was enabled before unless it was disabled explicitly with CLI option, so it was opt-out. Now you made it opt-in instead and I suggest to return to previous behavior and remove option that disabled sync from DSN so it is always enabled.